src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_InvalidExpression.vb

```diff
@@ -912,5 +913,49 @@ Block[B8] - Exit
             VerifyFlowGraphAndDiagnosticsForTest(Of MethodBlockSyntax)(source, expectedFlowGraph, expectedDiagnostics)
         End Sub
 
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem(35813, "https://github.com/dotnet/roslyn/issues/35813"), WorkItem(45382, "https://github.com/dotnet/roslyn/issues/45382")>
+        Public Sub InvalidTypeArguments()
+            Dim source = <![CDATA[
+Class C
+    Public Sub M(node As Object)
+        node.ExtensionMethod(Of Object)() 'BIND:"node.ExtensionMethod(Of Object)()"
+    End Sub
+End Class
+]]>.Value
+
+            Dim expectedStrictOperationTree = <![CDATA[
+IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid) (Syntax: 'node.Extens ... f Object)()')
+  Children(1):
+      IInvalidOperation (OperationKind.Invalid, Type: ?, IsInvalid) (Syntax: 'node.Extens ... (Of Object)')
+        Children(2):
+            IParameterReferenceOperation: node (OperationKind.ParameterReference, Type: System.Object, IsInvalid) (Syntax: 'node')
+            IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: '(Of Object)')
```

Comment:
>IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid) (Syntax: '(Of Object)') [](start = 12, length = 88)

Would it make sense to also filter out BoundTypeArguments in CreateBoundBadExpressionOperation?

Reply:
I waffled between that approach and creating a node for them, and ended up deciding on the approach I took because it felt like that was more correct: there's a number of odd things that can be the child of a BoundBadExpression, and BoundTypeArguments don't feel special in that regard. That being said, if you feel that we should instead just filter them out I'm not opposed to doing so.

---
In reply to: [568159440](https://github.com/dotnet/roslyn/pull/50942#discussion_r568159440) [](ancestors = 568159440)